### ISSUE TITLE
Showing Today for 1 day excludes entries after midnight GMT

### ISSUE THREAD

**Issue by cmause** [FamousWolf/week-planner-card#77]
When displaying just today on a dashboard, only items from midnight local through 5:00 p.m. local are displayed.  My time zone is GMT -700, so the cutoff corresponds to midnight GMT.

```
type: custom:week-planner-card
calendars:
  - entity: calendar.chris
    color: '#1976d2'
timeFormat: t
days: 1
startingDay: today
```

If I start playing with the days parameter, setting it to 1.2, for example, the full day will be displayed, but it will also display Tomorrow, with no events.

Desired behavior:  Displaying Today for 1 day should cover all events from midnight to midnight local time.
Additional adjustment:  Use only integer portion of Days parameter

**Comment by FamousWolf** [comment]
This sounds like a timezone problem. I'm not able to reproduce it though. Is this also a problem for you with other calendar cards?

**Comment by cmause** [comment]
When I show the weekly planner card for the whole week, Today works fine.  The problem only appears when I just show Today for just 1 day.

My time zone is GMT-0700 (Arizona time) for both my HA instance and my Google Calendar.  I have attached some images showing test calendar entries.  You can see the entries at 5:00 p.m. or later appear in the HA native calendar, but not on the weekly planner card.

I have also attached an image of the full week, showing Today goes past 5:00 p.m., as we would expect.  I haven't tested it, but I suspect the last day of any period is likely cut off at 5:00 p.m.

I would agree with your suspicion that it's a time zone problem.  Maybe not applying the time zone offset on the end of the date/time range?

I'm happy to provide an information that would help you troubleshoot.


![FullWeekCalendarCard](https://github.com/FamousWolf/week-planner-card/assets/29841207/a7cc18e6-3316-44b6-b15a-616d1576bbec)
![WeeklyPlannerCard](https://github.com/FamousWolf/week-planner-card/assets/29841207/73bc6da0-1119-48b0-ae07-5dbdbe2e6074)
![HACalendar](https://github.com/FamousWolf/week-planner-card/assets/29841207/47479acf-b11f-4eb8-a7ba-095f1eddf100)
![GoogleCalendar](https://github.com/FamousWolf/week-planner-card/assets/29841207/4bd32bd4-6920-4f4e-b246-240d4ca15852)

**Comment by cmause** [comment]
@FamousWolf , it appears you are in the Netherlands (GMT+0200).  Perhaps you could try creating an event at 1:00 a.m. your local time and see if it appears on your card?

I'm not a JavaScript developer, and I don't have a proper development environment (I was a .Net developer 20 years ago).  But looking through the code, I'm wondering if the call on line 451 in card.js is the culprit.  I see the 'Z' being added to specify Zulu (GMT) time, when the calendar entity appears to default to the local time zone.

**Comment by FamousWolf** [comment]
I'm able to reproduce it now. You could be right about the culprit. I'll test it today and hopefully I'll be able to fix it quickly

**Comment by FamousWolf** [comment]
You were right. I've fixed the problem and released version 1.5.1.

**Comment by cmause** [comment]
I have installed 1.5.1 and confirmed the fix worked.

Thanks @FamousWolf!  You're awesome!
